infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutContextData.kt

```diff
@@ -29,6 +33,8 @@ sealed class LayoutAsset<T : LayoutAsset<T>>(contextData: LayoutContextData<T>)
 
     @get:JsonIgnore
     abstract val contextData: LayoutContextData<T>
+
+    abstract fun withContext(contextData: LayoutContextData<T>): T
```

Comment:
Mites tämän käytännön käyttö suhteessa ulkopuoliseen maailmaan tulee menemään, onko tämä tarkoitettu puhtaasti sisäiseen käyttöön vai kutsutaanko tätä joskus muualtakin? Lähinnä siis mietin että onko tämä tarkoitettu palvelemaan puhtaasti kaikkia `asMainDraft()`/`asDesignOfficia()` jne. variantteja, vai halutaanko tätä kutsua joskus käsinkin. Lähinnä tämä vaikuttaa nimeämisiin, silä ensimmäisessä keississä nimeäisin tämän ehkä tyyliin `withContextInternal()` merkkaamaan sitä että älä kutsu ulkoa ja toisessa jotenkin tyyliin `copyWithContext()` merkkaamaan sitä, että tämä on selkeästi `copy()`-funkkari

Reply:
Joo, kyllä asMain/Design jne:t on ainoat jotka näitä käyttää, mutta sellaisenaankin, se tulee näiden olioiden ulkopuolelta, joten internal on vähän hassu nimitys. Sinällään en näe mitään syytä miksei noita voisi käyttää muualtakin jos kontekstinvaihtoa jostain syystä tehtäisi muualla.